core/txpool/legacypool/legacypool.go

```diff
@@ -883,6 +883,17 @@ func (pool *LegacyPool) promoteTx(addr common.Address, hash common.Hash, tx *typ
 		pendingDiscardMeter.Mark(1)
 		return false
 	}
+
+	// Check balance for overdraft
```

Comment:
If we directly check the total cost of the pending and queued list, it may cause some queued transactions to be deleted incorrectly.  
For example, a sender Alice sends 3 transactions, first, tx_1 deposits 10 ether to her account, then tx_2 transfers 5 Ether to Bob from Alice, and finally, tx_3 transfers 3 Ether to Bob again. If tx_2 and tx_3 come to the queued pool first without tx_1, the check of the total cost of the pending and queued list will delete those two without waiting for the first transaction tx_1.

Reply:
No, these will not happen because the queued list now does not check for latent overdrafts of tx. Therefore, the last two txs can be admitted to the queued pool first, and when the first tx reaches the pending pool, the promote function moves the two previously arrived txs to the pending pool. The movement also does not check for latent overdrafts of tx. So in our code, we add the check in the promote function.

Reply:
Hi, we think these two methods can both mitigate the attack and we are fine with implementing a check of latent overdraft in the queued list.